allennlp/interpret/attackers/input_reduction.py

```diff
@@ -0,0 +1,145 @@
+                    remove_one_token(grads[grad_input_field],
+                                     current_instance,
+                                     input_field_to_attack, ignore_tokens)
+
+            final_tokens.append(current_tokens)
+        return sanitize({"final": final_tokens, "original": original_tokens})
+
+def remove_one_token(grads: np.ndarray = None,
+                     instance: Instance = None,
+                     input_field_to_attack: str = 'tokens',
+                     ignore_tokens: List[str] = ["@@NULL@@"]) -> Tuple[Instance, int]:
+    """
+    Finds the token with the smallest gradient and removes it.
+    """
+    # Compute L2 norm of all grads.
+    grads_mag = [np.sqrt(grad.dot(grad)) for grad in grads]
+
+    # Skip all ignore_tokens by setting grad to infinity
+    text_field: TextField = instance[input_field_to_attack]  # type: ignore
+    for tok_idx, tok in enumerate(text_field.tokens):
+        if tok in ignore_tokens:
+            grads_mag[tok_idx] = float("inf")
+
+    # For NER, skip all tokens that are not in outside
+    if "tags" in instance:
+        tag_field: ListField = instance['tags']  # type: ignore
+        field_list = tag_field.field_list
+        for idx, label in enumerate(field_list):
+            label_field: LabelField = label  # type: ignore
+            if label_field.label != "O":
+                grads_mag[idx] = float("inf")
+
+    smallest = np.argmin(grads_mag)
+    if smallest == float("inf"): # if all are ignored tokens, return.
+        return instance, smallest
+
+    # remove smallest
+    input_field: TextField = instance[input_field_to_attack]  # type: ignore
+    inputs_before_smallest = input_field.tokens[0:smallest]
+    inputs_after_smallest = input_field.tokens[smallest + 1:]
+    input_field.tokens = inputs_before_smallest + inputs_after_smallest
+
+    if "tags" in instance:
+        field_list_before_smallest = field_list[0:smallest]
+        field_list_after_smallest = field_list[smallest + 1:]
+        tag_field.field_list = field_list_before_smallest + field_list_after_smallest
+
+    instance.indexed = False
+    return instance, smallest
+
+def get_ner_tags_and_mask(current_instance: Instance = None,
+                          input_field_to_attack: str = 'tokens',
+                          ignore_tokens: List[str] = ["@@NULL@@"]):
+    """
+    Used for the NER task. Sets the num_ignore tokens, saves the original
+     predicted tag and a 0/1 mask in the position of the tags
+    """
+    # Set num_ignore_tokens
+    num_ignore_tokens = 0
+    input_field = current_instance[input_field_to_attack]
+    for token in input_field.tokens:
+        if str(token) in ignore_tokens:
+            num_ignore_tokens += 1
+
+    # save the original tags and a 0/1 mask where the tags are
+    tag_mask = []
+    original_tags = []
+    tag_field = current_instance["tags"]
+    for label in getattr(tag_field, 'field_list'):
```

Comment:
I think using `getattr` to get around a mypy warning when you know the type of the object is awful.  Just do what we do above for `TextField`: `tag_field: SequenceLabelField = current_instance["tags"]  # type: ignore`.  You'll have to change the predictor to use the right type, here, too (I made a comment there).  Then this for loop becomes `for label in tag_field:`, and you don't need to call `label.label` below, just use `label`.